Marketplace: update notices and bubbles

raicem opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/Automattic/woocommerce.com/issues/23241.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout **trunk** and get a new build
2. Connect your store to WooCommerce.com but please uninstall WooCommerce.com Update Manager.
3. Checkout this branch and get a new build
4. First, visit the WooCommerce homepage: http://localhost:8888/wp-admin/admin.php?page=wc-admin
5. Confirm you see the **1** badge next to the **Extensions** submenu
6. Visit Discover, Extensions and Themes tabs. Confirm all have the WUM notice.
    - Visit Business services tab too to confirm it's **not** appearing there. 
8. Visit [My Subscriptions tab](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions) and then refresh the page
    - At this point we set  `my_subscriptions_tab_loaded` using `WC_Helper_Options::update`
    - To see the data structure, please do `cd plugins/woocommerce && nvm use && pnpm -- wp-env run cli wp option get woocommerce_helper_data`
    - If you'd like to set it to null, `cd plugins/woocommerce && nvm use && pnpm -- wp-env run cli wp shell` and then `WC_Helper_Options::update('my_subscriptions_tab_loaded', null);`. 
    - If you'd like to delete the key fully, again in `wp shell` run:
    ```
    $data = get_option( 'woocommerce_helper_data', array() );
    unset( $data['my_subscriptions_tab_loaded'] );
    update_option( 'woocommerce_helper_data', $data );
    ```
9. Confirm you no longer see the badge next to the **Extensions** submenu
10. Confirm you see a notice starting with "Please install the WooCommerce.com Update Manager" in the My Subscriptions tab.
11. Visit Discover, Extensions, Themes and Business Services tab and confirm you don't see the WooCommerce.com Update Manager notice.  
12. Delete the `my_subscriptions_tab_loaded` Helper option (following the command snippets in step 6). 
13. Load the Extensions page and confirm you see the notice again. On the notice, click the **download** button to download the ZIP. Manually update the plugin file but please don't activate it. 
14. Visit the Extensions tab again and confirm the notice is still up but with a different message, urging to activate it. 
15. Activate WUM and visit the Extensions tab again. Confirm you don't get a notice on the page and there is no bubble next to the Extensions sub menu item.
17. Please test around this issue

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Update notices and badges for WooCommerce.com Update Manager
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

raicem committed:
Marketplace: add a space before Marketplace Extensions bubble

raicem committed:
Markeplace: remove Extensions submenu bubble when My Subscriptions tab is visited

raicem committed:
Marketplace: show WUM notice only on My Subscriptions tab

raicem committed:
Marketplace: change where we detect My Subscriptions tab load

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

andfinally reviewed (commented):
Neat! Works nicely. My only slight issue is that we still want to show the `PluginInstallNotice` on the other tabs if the user hasn't visited My Subscriptions – is there some way we could do that?

(The `PluginInstallNotice` also appears if you just disable WUM rather than deleting it, which is what we want.)

raicem committed:
Marketplace: update logic to show the WUM notice

In a previous iteration I have misunderstood the assignment. We actually
would like to show the WUM notice on all tabs, to be removed after user
visits the My Subscriptions tab.

Comment:
https://github.com/woocommerce/woocommerce/pull/55657#pullrequestreview-2632544972

andfinally reviewed (approved):
Thanks! Works as expected.

I've just noticed the focus state of the buttons on the notice bumps together.

<img width="415" alt="image" src="https://github.com/user-attachments/assets/ccc9d343-662d-4a60-a416-fa2930d92ad8" />

We can improve it with this tweak:

```diff
Index: plugins/woocommerce/client/admin/client/marketplace/components/woo-update-manager-plugin/woo-update-manager-plugin.scss
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/plugins/woocommerce/client/admin/client/marketplace/components/woo-update-manager-plugin/woo-update-manager-plugin.scss b/plugins/woocommerce/client/admin/client/marketplace/components/woo-update-manager-plugin/woo-update-manager-plugin.scss
--- a/plugins/woocommerce/client/admin/client/marketplace/components/woo-update-manager-plugin/woo-update-manager-plugin.scss	(revision 63232cee89ed1878718fd76d575953979c56c5bb)
+++ b/plugins/woocommerce/client/admin/client/marketplace/components/woo-update-manager-plugin/woo-update-manager-plugin.scss	(date 1740410836869)
@@ -35,6 +35,8 @@
 		}
 
 		.components-notice__buttons {
+			display: flex;
+			gap: 8px;
 			margin-top: 12px;
 		}
 	}
```

raicem committed:
Marketplace: update WUM notice gap

raicem commented:
Thank you for the review @andfinally! [Implemented](https://github.com/woocommerce/woocommerce/pull/55657/commits/466179a433147318566d32d6137ad517b9cd40f0) the patch you have sent.
